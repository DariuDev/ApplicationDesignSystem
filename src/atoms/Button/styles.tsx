import {  StyleSheet } from 'react-native';

export default StyleSheet.create({
    // ...
    btn: {
      elevation: 8,
      backgroundColor: "#009688",
      borderRadius: 10,
      paddingVertical: 10,
      paddingHorizontal: 12,
    },
    txt: {
      fontSize: 18,
      color: "#fff",
      alignSelf: "center",
    }
  });

