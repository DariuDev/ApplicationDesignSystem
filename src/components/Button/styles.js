import {  StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    // ...
    btn: {
      elevation: 8,
      backgroundColor: "#009688",
      borderRadius: 10,
      paddingVertical: 10,
      paddingHorizontal: 12,
      marginHorizontal:10,
    },
    txt: {
      fontSize: 18,
      color: "#fff",
      fontWeight: "bold",
      alignSelf: "center",
    //   textTransform: "uppercase"
    }
  });

export {styles};
