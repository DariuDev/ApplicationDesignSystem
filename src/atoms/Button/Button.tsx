import React from 'react';
import { Text, Pressable,PressableProps } from 'react-native';
 import  styles  from './styles';
import {ButtonType} from '../../types';

// type ButtonTypeProops=ButtonType & PressableProps;
const Button= (
    {  isRightChildren=false ,children,onPress,
         title, styleText , styleButton ,
          testId = "button-wrap-test",...otherprops}:ButtonType & PressableProps) => {
    return (
            <Pressable  onPress={onPress} style={[ styles.btn,{...styleButton}]} testID={testId} {...otherprops}>
                {!isRightChildren &&  children}
                <Text style={[ styles.txt,{...styleText}]} testID='button-text-test'>{title}</Text>
                {isRightChildren && children}
            </Pressable>

    )
}
export default Button;

