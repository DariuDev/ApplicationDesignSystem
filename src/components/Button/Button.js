import React from 'react';
import { Text, Pressable, View } from 'react-native';
 import { styles } from './styles';


const Button = ({ onPress=()=>{}, title, styleText = {}, styleBtton = {} }) => {
    return (
        // <View>
            <Pressable onPress={onPress} style={[ styles.btn,{...styleBtton}]}>
                <Text style={[ styles.txt,{...styleText}]}>{title}</Text>
            </Pressable>
        // </View>

    )
}
export default Button;

