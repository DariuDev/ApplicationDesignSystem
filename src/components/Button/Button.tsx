import React from 'react';
import { Text, Pressable } from 'react-native';
 import  styles  from './styles';
import {ButtonType} from '../../types';


const Button:React.FC<ButtonType> = ({ onPress, title, styleText , styleButton , testId = "button-wrap-test"}) => {
    return (
            <Pressable onPress={onPress} style={[ styles.btn,{...styleButton}]} testID={testId}>
                <Text style={[ styles.txt,{...styleText}]} testID='button-text-test'>{title}</Text>
            </Pressable>

    )
}
export default Button;

