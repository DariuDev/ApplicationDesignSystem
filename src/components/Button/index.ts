import Button from "./Button";
export * from './styles';
export default Button;