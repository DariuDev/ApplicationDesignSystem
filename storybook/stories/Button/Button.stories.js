import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from '../CenterView';
import {Button} from '../../../src/atoms/Button';
import style from './styles';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

storiesOf('Btn ', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('Button 1', () => (
   
   <Button title={'Button 1'} styleBtton={style.btn1} isRightChildren={false} >
      <MaterialIcons name={'arrow-back-ios'} size={14} color={'pink'} />
   </Button>

  )).add('Button 2', () => (
   
    <Button title={'Button 2'}  styleBtton={style.btn2}/>
 
   )).add('Button 3', () => (
   
    <Button title={'Button 3'} styleBtton={style.btn3} isRightChildren={true}>
      <MaterialIcons name={'arrow-back-ios'} size={14} color={'pink'} />
    </Button>
 
   ))