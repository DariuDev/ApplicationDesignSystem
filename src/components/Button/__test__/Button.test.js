// import React from 'react';
// import renderer from 'react-test-renderer';
// import Button from '../Button';


// test('renders correctly Button', () => {
//   const tree = renderer.create(<Button title={'clickMe'}/>).toJSON();
//   expect(tree).toMatchSnapshot();
// });

import React from 'react';
import Button from '../Button';
import { fireEvent, render, wait } from '@testing-library/react-native';

const caption = 'Test button';
describe('Button', () => {
  const activeButton  = render(<Button title={caption}/>);
   const { getByTestId,  getByText,debug }= activeButton ;
  it('renders correctly', () => {
    const activeButtonAsJson = activeButton.toJSON();
    expect(activeButtonAsJson).toMatchSnapshot();

    expect(getByText(caption)).toBeTruthy();
    fireEvent.press(getByTestId("custtomButton"));

  });
  // expect(component).toMatchSnapshot();
  // it('Outputs the correct text', () => {
  //   expect(getByText(caption)).toBeTruthy();
  // });
  // expect(getByTestId("custtomButton")).toBeTruthy();

  // it('Fires the assigned event on press', () => {
  //   fireEvent.press(activeButton.getByTestId("custtomButton"));
    
  //     // expect().
  
  // });
  // fireEvent.press(getByTestId('custtomButton'))
});






