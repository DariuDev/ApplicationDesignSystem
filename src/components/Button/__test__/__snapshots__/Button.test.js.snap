// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#009688",
        "borderRadius": 10,
        "elevation": 8,
        "marginHorizontal": 10,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
      },
      Object {},
    ]
  }
  testID="custtomButton"
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
        },
        Object {},
      ]
    }
  >
    Test button
  </Text>
</View>
`;
